src/openms/include/OpenMS/DATASTRUCTURES/String.h

```diff
@@ -524,6 +527,25 @@ namespace OpenMS
       return false;
     }
 
+    bool operator==(const StringView other) const
+    {
+      if (size_ != other.size_) return false;
+
+      //same size
+      // same sequence, if both Views point to the same start
+      if (begin_ == other.begin_) return true;
+
+      const char * b = begin_;
```

Comment:
you can do this with less code: `strcmp(...)`
http://www.cplusplus.com/reference/cstring/strcmp/